src/Sarif.FunctionalTests/v2/ConverterTestData/PREfast/Expected-049.xml.sarif

```diff
@@ -12,20 +12,24 @@
           "mimeType": "text/x-cpp"
         }
       },
-      "logicalLocations": {
-        "": {
+      "logicalLocations": [
+        {
+          "fullyQualifiedName": "",
           "decoratedName": ""
         },
```

Comment:
The spec doesn't prohibit empty names, but... how did PREfast produce this?

Reply:
TODO. We should look at the logical for all logical location processing things: android studio, fxcop and prefast converters. Note that updating test files for these is tricky. If we change the code logical, new files won't match old. If we universally update all these older test files, we will no longer use the prerelease transformation logic to verify them. And so I suggest we defer these issues until we're ready for that wholesale conversion.

[https://github.com/Microsoft/sarif-sdk/issues/1171](https://github.com/Microsoft/sarif-sdk/issues/1171)

---
In reply to: [243852202](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243852202) [](ancestors = 243852202)